velox/common/base/AsyncSource.h

```diff
@@ -110,6 +110,7 @@ class AsyncSource {
         }
         std::swap(make, make_);
       }
+      resultMoved_ = true;
```

Comment:
What is the purpose of this variable? Which case does this handle?

Reply:
This variable ensures if `move` function is called, we don't need to execute the logic in `close` as all resources are already released / returned.
Thanks.

Reply:
Should this not be at the top then? For example, if the `item_` is present, we move and return above. This will not be set and we will execute `close()`

Reply:
We also return if `promise_` is set.

Reply:
Yes make sense, let me move it.

Reply:
Thanks! Can we name this `moved_`? Since we always don't move the result. For example, when the `promise_` is set.

Reply:
Upated, thanks.